### PR TITLE
check for outdated protofiles

### PR DESCRIPTION
**Pull request by agbpatro**

### CODE REVIEW

**Review by mrancourt — APPROVED** [review]

**Review by ThomasAlxDmy — APPROVED** [review]

File: go.mod [hunk 1]
```diff
@@ -36,7 +37,6 @@ require (
 	github.com/googleapis/gax-go/v2 v2.8.0 // indirect
 	github.com/jmespath/go-jmespath v0.4.0 // indirect
 	github.com/matttproud/golang_protobuf_extensions v1.0.4 // indirect
-	github.com/pebbe/zmq4 v1.2.10 // indirect
```
Comment (agbpatro) [inline go.mod:39]: zmq is not an indirect dependency

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -17,10 +17,36 @@ jobs:
...
-        make format
+        wget https://github.com/protocolbuffers/protobuf/releases/download/v25.0/protoc-25.0-linux-x86_64.zip
+        unzip protoc-25.0-linux-x86_64.zip
+        sudo mv bin/protoc /usr/local/bin/protoc
+        sudo mv include/* /usr/local/include/
+
+    - name: Install protoc-gen-go
+      run: go install google.golang.org/protobuf/cmd/protoc-gen-go@v1.28.1
+
+    - name: Generated protofiles are upto date
```
Comment (vmallet) [inline .github/workflows/build.yml:30]: `upto` -> `up to`

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -17,10 +17,36 @@ jobs:
...
+      run: |
+        make generate-golang
         git diff --exit-code
```
Comment (vmallet) [inline .github/workflows/build.yml:33]: What's the point of running this? Just for the logs?

**Review by vmallet — COMMENTED** [review]
See comments

File: .github/workflows/build.yml (hunk 3 above)
Comment (agbpatro) [reply .github/workflows/build.yml:33]: to ensure proto changes are kept in sync with generated files

File: .github/workflows/build.yml (hunk 2 above)
Comment (agbpatro) [reply .github/workflows/build.yml]: addressed

File: .github/workflows/build.yml (hunk 3 above)
Comment (agbpatro) [reply .github/workflows/build.yml:33]: updated. just keeping `git diff --exit-code` for now

File: README.md [hunk 4]
```diff
@@ -153,7 +153,7 @@ Prometheus or a StatsD interface supporting data store for metrics. This is requ
 ## Protos
 Data is encapsulated into protobuf messages of different types. We do not recommend making changes, but if you need to recompile them you can always do so with:
 
-  1. Install protoc, currently on version 3.21.12: https://grpc.io/docs/protoc-installation/
+  1. Install protoc, currently on version 25.0: https://grpc.io/docs/protoc-installation/
```
Comment (aaronpkahn) [inline README.md:156]: should this be `4.25.0`?

Comment (agbpatro) [reply README.md]: Updated. The download URL starting version 4 became a little different `https://github.com/protocolbuffers/protobuf/releases/download/v25.0/protoc-25.0-linux-x86_64.zip` hence I got confused

**Review by aaronpkahn — APPROVED** [review]

**Pull request merged**